src/antsibull/build_ansible_commands.py

```diff
@@ -397,6 +397,84 @@ def _extract_python_requires(
     )
 
 
+# pylint: disable-next=too-many-branches
+def feature_freeze_version(spec: str, collection_name: str) -> str:  # noqa: C901
+    """
+    Make semantic version range specification valid for feature freeze.
+    """
+    spec_obj = SemVerSpec(spec)
+
+    potential_clauses = []
+    upper_operator = None
+    upper_version = None
+    min_version = None
+    pinned = False
+
+    # If there is a single clause, it's available as spec_obj.clause;
+    # multiple ones are available as spec_obj.clause.clauses.
+    try:
+        clauses = spec_obj.clause.clauses
+    except AttributeError:
+        clauses = [spec_obj.clause]
+
+    # Look at each clause of the version specification
+    for clause in clauses:
+        if clause.operator in ("<", "<="):
+            if upper_operator is not None:
+                raise ValueError(
+                    f"Multiple upper version limits specified for {collection_name}: {spec_obj}"
```

Comment:
Would it be better to use the original user provided `spec` instead of the one that's been coerced by semantic_version?

```suggestion
                    f"Multiple upper version limits specified for {collection_name}: {spec}"
```

Reply:
Changed in 9e876254202688a0e0f418ff25a3a7730d8a2129.